x-pack/legacy/plugins/siem/public/components/open_timeline/delete_timeline_modal/index.tsx

```diff
@@ -4,58 +4,52 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { EuiButtonIcon, EuiModal, EuiToolTip, EuiOverlayMask } from '@elastic/eui';
-import React, { useCallback, useState } from 'react';
+import { EuiModal, EuiOverlayMask } from '@elastic/eui';
+import React, { useCallback } from 'react';
+import { createGlobalStyle } from 'styled-components';
 
 import { DeleteTimelineModal, DELETE_TIMELINE_MODAL_WIDTH } from './delete_timeline_modal';
-import * as i18n from '../translations';
 import { DeleteTimelines } from '../types';
+const RemovePopover = createGlobalStyle`
+div.euiPopover__panel-isOpen {
+  display: none;
+}
+`;
 
 interface Props {
-  deleteTimelines?: DeleteTimelines;
-  savedObjectId?: string | null;
-  title?: string | null;
+  deleteTimelines: DeleteTimelines;
+  onComplete?: () => void;
+  isModalOpen: boolean;
+  savedObjectIds?: string[] | null | undefined;
+  title: string | JSX.Element | null;
 }
 /**
  * Renders a button that when clicked, displays the `Delete Timeline` modal
  */
-export const DeleteTimelineModalButton = React.memo<Props>(
-  ({ deleteTimelines, savedObjectId, title }) => {
-    const [showModal, setShowModal] = useState(false);
-
-    const openModal = useCallback(() => setShowModal(true), [setShowModal]);
-    const closeModal = useCallback(() => setShowModal(false), [setShowModal]);
-
+export const DeleteTimelineModalOverlay = React.memo<Props>(
+  ({ deleteTimelines, isModalOpen, savedObjectIds, title, onComplete }) => {
+    const internalCloseModal = useCallback(() => {
+      if (onComplete != null) {
+        onComplete();
+      }
+    }, [onComplete]);
     const onDelete = useCallback(() => {
-      if (deleteTimelines != null && savedObjectId != null) {
-        deleteTimelines([savedObjectId]);
+      if (deleteTimelines != null && savedObjectIds != null) {
```

Comment:
I think deleteTimelines should always be there, so we can remove it from the condition.
Do we care is savedObjectIds is an empty array?